Sprint 3

EranosyanEduard opened the pull request:
Здравствуйте.

Для того, чтобы удалить чат или его пользователей необходимо кликать правой кнопкой мыши по определенному чату.
С тестированием роутера возникли трудности, поскольку необходимо подменять оригинальные интерфейсы с помощью пакета jsdom, работа которого отличается от оригинальных интерфейсов браузера (методы back и forward).

Перед запуском тестов - запустите скрипт "npm run before_test".

Я без ошибок запускаю скрипт "npm run start" и не понимаю почему тесты практикума не проходят.

EranosyanEduard committed:
fix: исправить парсинг пути маршрута

EranosyanEduard committed:
refactor: привести в соответствие тз пути маршрутов

EranosyanEduard committed:
feat: создать класс ApiClient - супер-класс для всех api

EranosyanEduard committed:
refactor: изменить названия code-style скриптов

EranosyanEduard committed:
fix: добавить проверку при возврате значения из ф-ии Router.findRoute

EranosyanEduard committed:
fix: изменить подход к именованию route-ов

EranosyanEduard committed:
fix: исправить типизацию и парсинг ответа сервера в ApiClient

EranosyanEduard committed:
feat: создать интерфейс для работы с auth api

EranosyanEduard committed:
feat: создать абстрактный класс Controller

EranosyanEduard committed:
feat: создать контроллер для работы с AuthApiClient

EranosyanEduard committed:
feat: добавить новый стиль для кнопки типа text

EranosyanEduard committed:
refactor: внести изменения в props-ы и типизацию RedirectButton

EranosyanEduard committed:
fix: исправить code-style в классе Component

EranosyanEduard committed:
fix: исправить значение props-а routeName в классе HomePage

EranosyanEduard committed:
feat: реализовать авторизацию пользователя

EranosyanEduard committed:
feat: реализовать регистрацию пользователя

EranosyanEduard committed:
refactor: заменить тип TListener на CallableFunctions в классе EventBus

EranosyanEduard committed:
feat: добавить импорт типа TOptions из компонента

EranosyanEduard committed:
feat: создать абстрактный класс Store

EranosyanEduard committed:
feat: реализовать хранилище данных о пользователе

EranosyanEduard committed:
feat: добавить ф-ю (декоратор) хранилища данных к классу Avatar

EranosyanEduard committed:
refactor: переименовать компонент UserBioCard в Settings

EranosyanEduard committed:
feat: добавить поддержку cookie в класс HttpClient

EranosyanEduard committed:
fix: исправить code-style в css компонента Button

EranosyanEduard committed:
fix: исправить импорт компонента Settings

EranosyanEduard committed:
fix: исправить работу хука Component.didMount

Добавить поведение, аналогичное поведению хуков didUpdate и willUpdate
жизненного цикла компонента, т.е. родительский компонент после монтирования
(didMount) инициирует у дочерних компонентов аналогичное событие.

EranosyanEduard committed:
refactor: изменить ф-ю apiClient.auth.getUser

EranosyanEduard committed:
fix: исправить импорт стилей компонента Settings

EranosyanEduard committed:
fix: исправить code-style шаблона html-страницы

EranosyanEduard committed:
fix: исправить работу ф-ии-декоратора useState

EranosyanEduard committed:
feat: добавить поддержку api на странице Settings

EranosyanEduard committed:
feat: добавить контроль за состоянием пользователя при навигации

EranosyanEduard committed:
feat: добавить в класс Store возможность хранения его копии

EranosyanEduard committed:
feat: создать интерфейс для работы с user api

EranosyanEduard committed:
feat: создать контроллер для работы с UserApiClient

EranosyanEduard committed:
refactor: исключить из хранилища User свойство userId

EranosyanEduard committed:
fix: исправить типизацию свойства body в TReqOptions

EranosyanEduard committed:
feat: реализовать изменение учетных данных пользователя

EranosyanEduard committed:
refactor: изменить название типа TComponentOptions на TComponentOpts

EranosyanEduard committed:
feat: комит - свалка фич

EranosyanEduard committed:
fix: fix code style

EranosyanEduard committed:
fix: исправить тип переменной, хранящей ссылку на интервал

EranosyanEduard committed:
feat: добавить поддержку тестов

EranosyanEduard committed:
refactor: удалить лишнее из tsconfig

EranosyanEduard committed:
feat: создать тесты для класса View

EranosyanEduard committed:
fix: исправить типизацию HttpClient-а

EranosyanEduard committed:
feat: создать тесты для HttpClient-а

EranosyanEduard committed:
feat: создать тесты для класса Router

EranosyanEduard committed:
refactor: вынести accessors в свойства классов

EranosyanEduard committed:
feat: комит - свалка фич

EranosyanEduard committed:
feat: обновить README.md

EranosyanEduard committed:
fix: исправить скрипт в package.json

EranosyanEduard commented:
Перед запуском тестов - запустите скрипт "npm run before_test".

podlovchenko reviewed (changes requested):
Привет, интересная архитектура приложения)
Есть замечания по работе системы:
1. Нет редиректа на страницу чатов, если пользователь в системе
2. При обновление страницы Cannot GET
3. Не получилось загрузить аватар

EranosyanEduard committed:
feat: добавить редирект авторизованного пользователя

Добавить автоматический редирект авторизованного пользователя при
повторном входе в приложение со страницы авторизации на страницу с
чатами.

EranosyanEduard committed:
fix: исправить навигацию на сервере

EranosyanEduard committed:
fix: исправить работу запроса, отправляющего файл аватара

EranosyanEduard committed:
fix: исправить слияние атрибутов

EranosyanEduard committed:
fix: удалить неиспользуемые файлы

EranosyanEduard committed:
feat: добавить интерфейс objButNotArr в класс Is

EranosyanEduard committed:
feat: добавить интерфейс diffKeys в класс ObjMeths

EranosyanEduard committed:
feat: добавить интерфейс diffKeys в класс ObjMeths

EranosyanEduard committed:
Merge remote-tracking branch 'origin/sprint_3' into sprint_3

# Conflicts:
#	src/utils/meths/ObjMeths.ts

EranosyanEduard committed:
fix: исправить установку значения в классе Store, если значение - объект

podlovchenko commented:
Привет, не могу войти в систему, ответ от сервера `User already in system`, а редиректа на страницу чатов нет, но со страницы чатов есть редирект на форму входа

podlovchenko commented:
Я запускала через `npm run serve`
Через `npm run start` работает
Переоткрой, я приму работу
Но нужно будет разобраться с командами и запусками в dev режиме (здесь достаточно parcel serve, посмотри почему не работает редирект, или parcel build и nodemon)  и в режиме продакшена (здесь parcel build и node)

EranosyanEduard commented:
Ничего не менял, отправил повторно, как вы и просили.

EranosyanEduard committed:
Merge branch 'main' into sprint_3
